docs/docs/developers/tutorials/codealong/first_fees.md

```diff
@@ -0,0 +1,354 @@
+
+When a PXE creates a Schnorr account, there are three key things that occur:
+ - generation of keys privately
+ - registeration of a Schnorr account locally (effectively an instance of the Schnorr account class)
+   - at this stage the address of the account is known, even though it is not yet deployed
+ - initialization and deployment of the account contract to the network
+
+**Create the keys and account in the PXE**
+
+Lets first create the account but only register the instance in the pxe. Its address will be calculated, and in a later step we can deploy it counterfactually to the network.
+
+```bash
+aztec-wallet create-account --register-only -a main
+```
+
+The `-a main` sets this new account's alias to, "main". Use `aztec-wallet get-alias` (with no params) to see all aliases.
+
+The equivalent using aztec.js - create a random account locally (we will deploy it in the next step):
+
+```javascript
+import { Fr } from "@aztec/aztec.js";
+import { getSchnorrAccount } from '@aztec/accounts/schnorr';
+async function main() {
+  let secretKey = Fr.random(); let salt = Fr.random();
+  let schnorrAccount = await getSchnorrAccount(pxe, secretKey, deriveSigningKey(secretKey), salt);
+}
+```
+
+Your PXE now has keys for an account that can be deployed on Aztec network(s).
+
+## Paying for an account deployment transaction
+
+To make transactions on the network, your account contract will need to specify a payment method of the enshrined asset, "Fee Juice".
+
+### Sponsored FPC
+
+To bootstrap first use, a sponsored fee paying contract (the canonical sponsored FPC) can be used to deploy your first account.
+
+**TODO: as snippet**
+A fee paying contract (FPC) effectively implements fee abstraction. It is a contract that pays for transactions of other accounts, when its own custom criteria is met.
+In the case of the canonical sponsored FPC, the only criteria is an upper bound on how much it sponsors an account's transactions. This will be enough to at least deploy an account.
+
+The PXE can be queried for the canonical sponsored FPC address, and then specified as the payment method.
+
+Via the CLI:
+
+The alias set earlier can be confirmed using: `aztec-wallet get-alias accounts:main`, this is specified here in `--from main`.
+
+```bash
+SPONSORED_FPC_ADDRESS=`aztec-wallet --get-canonical-sponsored-fpc-address`
+aztec-wallet deploy-account --from main --payment method=fpc-sponsored,fpc=$SPONSORED_FPC_ADDRESS
+```
+
+The equivalent using aztec.js - get sponsored fpc address and use payment method:
+
+```javascript
+import { SponsoredFeePaymentMethod } from "@aztec/aztec.js/fee/testing";
+import { getDeployedSponsoredFPCAddress } from "../src/utils/sponsored_fpc.js"; // currently manually brought into project (see aztec-starter repo)
```

Comment:
I think you want to get `getSponsoredFPCInstance` like [here](https://github.com/AztecProtocol/aztec-starter/pull/80/files#diff-36d808495de81c0ef334263da01fb16ea4380b9d54c9ca49c3f445eb4f382650R74-R76). I think what you have will work in the sandbox, but not if a PXE is running separately.

Reply:
Updated.